self_enrollment_tool/views.py

```diff
@@ -217,44 +217,57 @@ def enable(request, course_instance_id):
     if start_date and end_date and end_date < start_date:
         messages.error(request, "End date cannot be before the start date.")
         return redirect('self_enrollment_tool:add_new')
-    
+
     try:
         if str(role_id) and course_instance_id:
             try:
-                course_instance = SelfEnrollmentCourse.objects.get(course_instance_id=course_instance_id, role_id=role_id)
-                logger.error(f'Self Enrollment is already enabled for this course  {course_instance_id} ')
-                messages.error(request, f'Self Enrollment is already enabled for this course (SIS ID {course_instance_id}) and role ({role_name}). '
-                              f'See below for the previously-generated link.')
-                path = reverse('self_enrollment_tool:enroll', args=[course_instance.uuid])
-            except SelfEnrollmentCourse.DoesNotExist:
-                uuid = str(uuid4())
-                SelfEnrollmentCourse.objects.create(course_instance_id=course_instance_id,
-                                              role_id=role_id,
-                                              updated_by=str(request.user),
-                                              uuid=uuid,
-                                              start_date=start_date,
-                                              end_date=end_date)
-                logger.debug(f'Successfully saved Role_id {role_id} for Self Enrollment in course {course_instance_id}. UUID={uuid}')
-                messages.success(request, f"Generated self-registration link. See details below.")
-
-            # install the self-unenroll tool
-            self_unenroll_client_id = settings.SELF_UNENROLL_CLIENT_ID
-            try:
-                result = install_unenrollment_tool(course_instance_id=course_instance_id, client_id=self_unenroll_client_id)
-                if result == 'installed':
-                    messages.success(request, f'Successfully installed self-unenrollment tool into the Canvas course.')
-                elif result == 'already_installed':
-                    messages.success(request, f'Self-Unenrollment tool already installed into the Canvas course.')
+                course_instance = SelfEnrollmentCourse.objects.filter(
+                    course_instance_id=course_instance_id,
+                    role_id=role_id,
+                    start_date=start_date,
+                    end_date=end_date
```

Comment:
Just want to clarify which direction we chose and if this should be an is_null check or not

Reply:
we will be going in the direction of is_null check